Phpthumb update

Mischosch opened the pull request:
Hey there,

updated phptumb usage to last available version 2.0 (https://github.com/masterexploder/PHPThumb/releases).

I added settings for:
- imagemagick_path
- quality_small
- quality_mini

This enables to have much better picture quality then with gd and 70 quality setting hardcoded.

Feel free to review and give feedback. If there are things to be fixed, let me know.


Cheers & Regards (great app!)
Mischosch

thibaud-rohmer commented:
Hi there !

Seems great, and I'd love to accept your pull request, but I don't want the config.php file to be changed (this will create conflicts for all users, for they edited this file already to put the path to their picturs directory).

Parameters that can be configured must be in src/classes/settings.php

If you want, though, you can just update phpthumb and set your options to default parameters, and I'll edit that code afterwards to support user settings.

But really, most important, please make sure to submit a pull request without a new config.php file :)

Mischosch commented on src/classes/Provider.php:
```diff
@@ -151,82 +151,109 @@ public static function Video($file){
+            $thumb = new phpthumb();
+            if (!empty(Settings::$imagemagick_path)) {
+                $thumb->config_imagemagick_path = Settings::$imagemagick_path;
+            }
```
if we do provide default settings through Setting.php setting, this will triggered and phpthumb will go into imagick mode… not sure how it behaves, if no imagick installation exists. will test this…

Mischosch commented:
Hey, removed config.php changes and put defaults into Settings.php. I tested what happens when imagick path is wrong/not existing - phpThumb does not error and goes the GD way. So I think this should be fine.

Added some infos to README related to the changes and how to change the defautl values. If you want any changes, like wording, feel free to change them.

thibaud-rohmer commented:
Perfect, thanks !
